src/main/java/org/mozilla/msrp/platform/vertical/video/YoutubeRepository.kt

```diff
@@ -0,0 +1,70 @@
+package org.mozilla.msrp.platform.vertical.video
+
+import com.google.api.services.youtube.YouTube
+import org.mozilla.msrp.platform.util.hash
+import org.mozilla.msrp.platform.util.logger
+import javax.inject.Inject
+import javax.inject.Named
+
+@Named
+class YoutubeRepository @Inject constructor(private val youtube: YouTube) {
+
+    @Inject
+    lateinit var youtubeApiInfo: YoutubeApiInfo
+
+    val log = logger()
+
+    fun videoList(query: String, locale: String, maxResult: Long): List<VideoItem>? {
+        return try {
+            log.info("[Youtube]====start retrieving list")
+            val searchRequest = youtube.search().list("id,snippet")
+            searchRequest.key = youtubeApiInfo.apiKey
+            searchRequest.q = query
+            searchRequest.type = "video"
+            searchRequest.order = "viewCount"
+            searchRequest.relevanceLanguage = locale
+            if (maxResult > 0) {
+                searchRequest.maxResults = maxResult
+            }
+
+            val searchResponse = searchRequest.execute()
+            val searchResultList = searchResponse.items
+            val videoList = if (!searchResultList.isNullOrEmpty()) {
+                // Both video duration and view count information cannot be retrieved from the search api.
+                // So, made another video list api call with all the video id from the previous search result
+                val idList: String = searchResultList.map { it.id.videoId }.reduce { accumulated, id -> "$accumulated,$id" }
+                val videoDetailRequest = youtube.videos().list("contentDetails,statistics")
```

Comment:
this will call the `videos()` API m*n times.
Where m is the number of the repository is used, and n is the number of the result set.
Can we cache n so we don't need to get the statistics repeatedly?

Reply:
As discussed, we don't need to cache this.